lagom-eventstore-example/lagom-eventstore-impl/src/main/java/com/lightbend/lagom/eventstore/impl/readside/EventStoreRepositoryImpl.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2016-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.eventstore.impl.readside;
+
+import akka.Done;
+import com.ibm.event.catalog.ResolvedTableSchema;
+import com.ibm.event.catalog.TableSchema;
+import com.ibm.event.common.ConfigurationReader;
+import com.ibm.event.oltp.EventContext;
+import com.ibm.event.oltp.EventContext$;
+import com.ibm.event.oltp.EventError;
+import com.ibm.event.oltp.InsertResult;
+import com.lightbend.lagom.eventstore.impl.writeside.HelloEvent;
+import com.typesafe.config.Config;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.RowFactory;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import scala.Option;
+import scala.collection.IndexedSeq;
+import scala.collection.JavaConversions;
+import scala.collection.Seq;
+import scala.compat.java8.FutureConverters;
+import scala.compat.java8.OptionConverters;
+
+import javax.inject.Singleton;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+@Singleton
+public class EventStoreRepositoryImpl {
```

Comment:
Unfortunately I can't run this code from within `mvn lagom:runAll` to test it. IBM's `ibm-event.jar` and all it's transitive dependencies need manual handling and we're currently stopped until that's handled via maven repos or ivy.

Any visual review of this code is more than welcome and will probably detect some immediate issues a simple run should have spotted.